lib/auth/authclient/authclient.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package authclient contains common code for creating an auth server client
+// which may use SSH tunnelling through a proxy.
+package authclient
+
+import (
+	"context"
+	"crypto/tls"
+
+	apiclient "github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/lib"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/reversetunnel"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+)
+
+// Config holds configuration parameters for connecting to the auth service.
+type Config struct {
+	// TLS holds credentials for mTLS.
+	TLS *tls.Config
+	// SSH is client SSH config.
+	SSH *ssh.ClientConfig
+	// AuthAddrs is a list of possible auth or proxy server addresses.
+	AuthServers []utils.NetAddr
+	// Log sets the logger for the client to use.
+	Log logrus.FieldLogger
+}
+
+// Connect creates a valid client connection to the auth service.  It may
+// connect directly to the auth server, or tunnel through the proxy.
+func Connect(ctx context.Context, cfg *Config) (auth.ClientI, error) {
+	cfg.Log.Debugf("Connecting to auth servers: %v.", cfg.AuthServers)
```

Comment:
Technically these might be proxies, not auth servers, right?

```suggestion
	cfg.Log.Debugf("Connecting to: %v.", cfg.AuthServers)
```